lib/teams/geneai-team/genai-team.ts

```diff
@@ -0,0 +1,66 @@
+import { ClusterInfo } from "../../spi";
+import { ApplicationTeam, TeamProps } from "../team";
+import { createNamespace } from "../../utils/namespace-utils";
+import { getBedrockPolicyDocument } from "./iam-policy";
+import * as iam from "aws-cdk-lib/aws-iam";
+
+/**
+ * Interface to define a GenAI on EKS team
+ */
+export interface GenAITeamProps extends TeamProps {
+    /**
+     * Name of the service account namespace.
+     */
+    namespace?: string;
+
+    /**
+     * Create Namespace with the provided one.
+     */
+    createNamespace?: boolean
+
+    /**
+     * Name of the service account for Bedrock.
+     */
+    serviceAccountName?: string;
+}
+
+/*
+ *This class define the Team that can be used with `GenAIBuilder`
+ *Sets IRSA for access to bedrock with required IAM policy along with creating a namespace.
+*/
+
+export class GenAITeam extends ApplicationTeam {
+
+  private genAITeam: GenAITeamProps;
+  /**
+   * @public
+   * @param {GenAITeamProps} props the Gen AI team definition {@link GenAITeamProps}
+   */
+  constructor(props: GenAITeamProps) {
+    super(props);
+    this.genAITeam = props;
+  }
+
+  setup(clusterInfo: ClusterInfo): void {
```

Comment:
Does it work? I see this code will clash with the setupServiceAccount method from the application team. I don't think you need to set it up like this at all. All you need to do is pass the serviceaccountname and serviceAccountPolicies to with the props to the Application team and its default setup method should take care of the IRSA stuff.